services/apps/portal/src/application/application-owner/application-owner.service.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../../user/user.entity';
+import { ApplicationParcelService } from '../application-parcel/application-parcel.service';
+import { Application } from '../application.entity';
+import { ApplicationOwnerType } from './application-owner-type/application-owner-type.entity';
+import {
+  ApplicationOwnerCreateDto,
+  ApplicationOwnerUpdateDto,
+} from './application-owner.dto';
+import { ApplicationOwner } from './application-owner.entity';
+
+@Injectable()
+export class ApplicationOwnerService {
+  constructor(
+    @InjectRepository(ApplicationOwner)
+    private repository: Repository<ApplicationOwner>,
+    @InjectRepository(ApplicationOwnerType)
+    private typeRepository: Repository<ApplicationOwnerType>,
+    private parcelService: ApplicationParcelService,
+  ) {}
+
+  async fetchByApplicationFileId(fileId: string) {
+    return this.repository.find({
+      where: {
+        application: {
+          fileNumber: fileId,
+        },
+      },
+      relations: {
+        type: true,
+      },
+    });
+  }
+
+  async create(createDto: ApplicationOwnerCreateDto, application: Application) {
+    const type = await this.typeRepository.findOneOrFail({
+      where: {
+        code: createDto.typeCode,
+      },
+    });
+
+    const parcel = await this.parcelService.getOneOrFail(createDto.parcelUuid);
+
+    const newOwner = new ApplicationOwner({
+      firstName: createDto.firstName,
+      lastName: createDto.lastName,
+      organizationName: createDto.organizationName,
+      email: createDto.email,
+      phoneNumber: createDto.phoneNumber,
+      application,
+      type,
+      parcels: [parcel],
+    });
+
+    return await this.repository.save(newOwner);
```

Comment:
I do not think you need to await return value

Reply:
I don't but adding / removing await impacts the errors / stack trace. Without await the error would be in the controller, with await the error will be in the service.

Reply:
got it, I think then we should probably have a rule regarding that. so the code looks and feels consistent